packages/x-charts-pro/src/context/ZoomProvider/initializeZoomData.ts

```diff
@@ -0,0 +1,16 @@
+import { ZoomState } from './Zoom.types';
+
+// This function is used to initialize the zoom data when it is not provided by the user.
+// It is helpful to avoid the need to provide the possibly auto-generated id for each axis.
+export const initializeZoomData = (
```

Comment:
Currently the user can use an empty array in the controlled zoom. We then fill it up correctly on the first user interaction.

Reply:
> I'm not sure we should enforce the correct usage from the start?

I don't see issue asking users to start with the correct behavior.

I assume most of the use-cases will be `useState([ { axisId: 'my-x-axis', start: 0, end: 100 } ])`

Reply:
From its usage it does not really look like an initializer.

I would expect that `initializeZoomData` is used to initialize the state. For example in the default value of the `useControlledProps`. But you use it at the beginning of every zoom interaction.

Either you consider it more as a cleaner function that ensure the axis are relevant with the options.
Then I would move the usage of this function in the  `value = React.useMemo(...` of the provider to call it only once.

Or it's an initializer and it should be called once in the `useControlledProps`.

Notice that this solution does not support the case of a user adding/removing the zoom option of an axis. It's an edge case so no need for it to be fixed. Maybe other a highlight in the docs such as

> You can not switch an axis from `zoom: false` to `zoom: true`.
> The current workaround is to control the `zoomData` and add this new zoomable axis to it.